Adding data creator functions

HrayrMuradyan opened the pull request:
# What is new?
3 new py files are added to the data folder of audiocraft.

- create_data.py
- description_generator.py
- essentia_utils.py

1) **Essentia utils:** Contains all functions connected to essentia for labeling audio.
2) **Description Generator:** Contains the description generator class that is used for generating descriptions from music attributes.
3) **Create Data:** Has all the necessary functions for creating the dataset.

See **Data Creator.ipynb** for step-by-step data creation.

# The structure of the Dataset folder

Have **Dataset** folder outside the audiocraft having:

- youtube_music_links folder -> train -> links.jsonl
- raw_music folder -> train -> .wav/.json (will be automatically extracted when running step-by-step data creation with Data Creator.ipynb)
- essentia_weights folder ->.pb weight files for Essentia


# The steps for data creation

1) Download_split = Downloads the .wav music files from youtube links provided in links.jsonl. As well as, saves temporal json files having information about which labeling technique should be used and who is the artist of the music.
2) Divide_into_clips = Divides the music into 30-second parts, duplicates the json files for each 30-second clip.
3) Prepare attributes = Overwrites the temporal json file with the main one with all the necessary keys (an empty structure that should be filled).
4) Fill_json_split = Fills the json file for each 30-second audio file.
5) Fill_Descriptions = Fills descriptions based on the json attributes.

Commit:
Adding data creator functions

KarlosMuradyan reviewed (commented):
Is there a reason of the change in `Data Preprocessor.ipynb`? Please remove it from the PR if the changes are not necessary

KarlosMuradyan commented:
Also would be good to add an example of `links.jsonl` file

KarlosMuradyan commented:
Can you add information of how to run from end-to-end all the notebooks to do the pre-processing and training? 
It's not directly related to this PR but will be super helpful

KarlosMuradyan commented:
The rest looks good. Great job!

Commit:
Added Generation notebook, combined data creation into one notebook, changed some variable names

KarlosMuradyan committed:
Made few changes in the code preprocessing and training:

- Fixed errors in data creation
- Fixed T5 encoding length
- Added collater function for batch training in memory_save mode
